### PR TITLE
brave: add kerberos authentication

### PR DESCRIPTION
**Pull request by Feyorsh**
###### Description of changes
Added dependency on `libkrb5` to allow for Kerberos authentication (e.g. SSO). This change is in line with [Chromium](https://github.com/NixOS/nixpkgs/blob/d77c6e5e690056d1212b93161225040208da0b7f/pkgs/applications/networking/browsers/chromium/common.nix#L147) and other browsers like [Firefox](https://github.com/NixOS/nixpkgs/blob/d77c6e5e690056d1212b93161225040208da0b7f/pkgs/applications/networking/browsers/firefox/common.nix#L91).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Feyorsh** [comment]
Bumping this. Idk if they have merge perms, but @buckley310 seems active?

**Comment by buckley310** [comment]
I do not have merge rights, unfortunately.

**Comment by Feyorsh** [comment]
@NickCao Would you mind taking a look, please? This is a fairly simple fix that fixes a feature regression unique to nixpkgs' version of Brave.

File: pkgs/applications/networking/browsers/brave/default.nix [hunk 1]
```diff
@@ -68,12 +69,11 @@ let
...
-    fontconfig freetype gdk-pixbuf glib gtk3 libdrm libX11 libGL
-    libxkbcommon libXScrnSaver libXcomposite libXcursor libXdamage
-    libXext libXfixes libXi libXrandr libXrender libxshmfence
-    libXtst libuuid mesa nspr nss pango pipewire udev wayland
-    xorg.libxcb zlib snappy
+    alsa-lib at-spi2-atk at-spi2-core atk cairo cups dbus expat fontconfig
+    freetype gdk-pixbuf glib gtk3 libGL libX11 libXScrnSaver libXcomposite
+    libXcursor libXdamage libXext libXfixes libXi libXrandr libXrender
+    libXtst libdrm libkrb5 libuuid libxkbcommon libxshmfence mesa nspr
+    nss pango pipewire snappy udev wayland xorg.libxcb zlib
```
Comment (NickCao) [inline pkgs/applications/networking/browsers/brave/default.nix:76]: May you drop this part to minimize the diff? (Or move it into its own commit)

File: pkgs/applications/networking/browsers/brave/default.nix [hunk 2]
```diff
@@ -111,6 +112,9 @@ stdenv.mkDerivation rec {
     # needed for GSETTINGS_SCHEMAS_PATH
     glib gsettings-desktop-schemas gtk3
 
+    # needed for GSSAPI authentication (Kerberos)
+    libkrb5
+
```
Comment (NickCao) [inline pkgs/applications/networking/browsers/brave/default.nix:117]: Adding it to `deps` should already be sufficient, no need for adding to `buildInputs`. Apart from this LGTM.

Comment (Feyorsh) [reply pkgs/applications/networking/browsers/brave/default.nix]: Tested and fixed it, although as a Nix noob I'm not sure what the role of `buildInputs` are if not for runtime dependencies (it looks like that's what `rpath` does in this context). In this context, are `buildInputs` just for making environment variables known to Brave?

Comment (NickCao) [reply pkgs/applications/networking/browsers/brave/default.nix]: In most cases buildInputs are for the compilers, while for this binary package there is no compiler involved, but the deps are directly used to construct rpath.
